specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/hyperdrive.json

```diff
@@ -0,0 +1,714 @@
+            "description": "Hyperdrive run id to cancel.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "header",
+            "name": "RunHistoryHost",
+            "description": "The host for run location.",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/HyperDriveCancelExperimentResponse"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/HyperDriveErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Cancel a HyperDrive experiment": {
+            "$ref": "./examples/cancelHyperDriveExperiment.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "HyperDriveErrorResponse": {
+      "properties": {
+        "code": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Error response code."
+        },
+        "message": {
+          "type": "string",
+          "description": "Error message."
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ],
+      "type": "object",
+      "description": "Response in case of an error."
+    },
+    "HyperDriveCancelExperimentResponse": {
+      "properties": {
+        "code": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Response code."
+        },
+        "result": {
+          "type": "string",
+          "description": "Message indicating operation success."
+        }
+      },
+      "required": [
+        "code",
+        "result"
+      ],
+      "type": "object",
+      "description": "Response for HyperDrive_CancelExperiment in case of success."
+    },
+    "HyperDriveExperimentBase": {
+      "description": "Base object for both request and response of HyperDrive_CreateExperiment api.",
+      "properties": {
+        "description": {
+          "maxLength": 511,
+          "type": "string",
+          "x-nullable": true,
+          "description": "The description for Hyperdrive run."
+        },
+        "generator_config": {
+          "description": "Hyperparameter space and the sampling method configuration.",
+          "properties": {
+            "name": {
+              "type": "string",
+              "enum": [
+                "RANDOM",
+                "GRID",
+                "BAYESIANOPTIMIZATION"
+              ],
+              "description": "Hyperparameter sampling method.",
+              "x-ms-enum": {
+                "name": "Name",
+                "modelAsString": true
+              }
+            },
+            "parameter_space": {
+              "type": "object",
+              "additionalProperties":{
+                "type": "array",
+                "items": {},
```

Comment:
I don't think you can express this API in swagger 2.0. And I'd have some suggestions on how to make the API better, but since the service is already published, that's probably not an option.

Reply:
ok.

Reply:
Postponed. Captured work for next preview.